[BUG]  qa_nightly_select_test.py::test_select integration test fails

**Describe the bug**
0.6 integration nightly build failed with:

04:06:12  FAILED src/main/python/qa_nightly_select_test.py::test_select[* WHERE strF= 'Yuan' OR intF=10][INCOMPAT, APPROXIMATE_FLOAT]


```
 =================================== FAILURES ===================================
04:06:12  _________________ test_select[* WHERE strF= 'Yuan' OR intF=10] _________________
04:06:12  
04:06:12  sql_query_line = ("SELECT * FROM test_table WHERE strF= 'Yuan' OR intF=10", "* WHERE strF= 'Yuan' OR intF=10")
04:06:12  pytestconfig = <_pytest.config.Config object at 0x7f96c84274f0>
04:06:12  
04:06:12      @approximate_float
04:06:12      @incompat
04:06:12      @qarun
04:06:12      @pytest.mark.parametrize('sql_query_line', SELECT_SQL, ids=idfn)
04:06:12      def test_select(sql_query_line, pytestconfig):
04:06:12          sql_query = sql_query_line[0]
04:06:12          if sql_query:
04:06:12              print(sql_query)
04:06:12              with_cpu_session(num_stringDf)
04:06:12  >           assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.sql(sql_query), conf=_qa_conf)
04:06:12  
04:06:12  src/main/python/qa_nightly_select_test.py:167: 
04:06:12  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
04:06:12  src/main/python/asserts.py:364: in assert_gpu_and_cpu_are_equal_collect
04:06:12      _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
04:06:12  src/main/python/asserts.py:356: in _assert_gpu_and_cpu_are_equal
04:06:12      assert_equal(from_cpu, from_gpu)
04:06:12  src/main/python/asserts.py:93: in assert_equal
04:06:12      _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
04:06:12  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
04:06:12  
04:06:12  cpu = [Row(strF='Yuan', byteF=20, shortF=600, intF=2200, longF=None, floatF=20.1200008392334, doubleF=2.000013, decimalF=2.01, booleanF=False, timestampF=datetime.datetime(2020, 2, 1, 12, 1, 1), dateF=datetime.date(1990, 1, 1))]
04:06:12  gpu = [Row(strF='take out', byteF=20, shortF=600, intF=None, longF=20, floatF=20.1200008392334, doubleF=2.000013, decimalF=2...3, decimalF=2.01, booleanF=False, timestampF=datetime.datetime(2020, 2, 1, 12, 1, 1), dateF=datetime.date(1990, 1, 1))]
04:06:12  float_check = <function get_float_check.<locals>.<lambda> at 0x7f96137558b0>
04:06:12  path = []
04:06:12  
04:06:12      def _assert_equal(cpu, gpu, float_check, path):
04:06:12          t = type(cpu)
04:06:12          if (t is Row):
04:06:12              assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
04:06:12              if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
04:06:12                  for field in cpu.__fields__:
04:06:12                      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
04:06:12              else:
04:06:12                  for index in range(len(cpu)):
04:06:12                      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
04:06:12          elif (t is list):
04:06:12  >           assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
04:06:12  E           AssertionError: CPU and GPU list have different lengths at [] CPU: 1 GPU: 2
04:06:12  
04:06:12  src/main/python/asserts.py:39: AssertionError
```

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala
@@ -146,25 +146,33 @@ object GpuFilter extends Arm {
     }
   }
 
+  private def allEntriesAreTrue(mask: GpuColumnVector): Boolean = {
+    if (mask.hasNull) {
+      false
+    } else {
+      withResource(mask.getBase.all()) { all =>
+        all.getBoolean
+      }
+    }
+  }
+
   def apply(batch: ColumnarBatch,
       boundCondition: Expression) : ColumnarBatch = {
     withResource(batch) { batch =>
-      val checkedFilterMap = withResource(
-        GpuProjectExec.projectSingle(batch, boundCondition)) { filterMap =>
-        // If  filter is a noop then return a None
-        withResource(filterMap.getBase.all()) { allIncluded =>
-          if (allIncluded.getBoolean) {
-            None
-          } else {
-            Some(filterMap.getBase.incRefCount())
-          }
+      val checkedFilterMask = withResource(
+        GpuProjectExec.projectSingle(batch, boundCondition)) { filterMask =>
+        // If  filter is a noop then return a None for the mask
+        if (allEntriesAreTrue(filterMask)) {
+          None
+        } else {
+          Some(filterMask.getBase.incRefCount())
         }
       }
-      checkedFilterMap.map { checkedFilterMap =>
-        withResource(checkedFilterMap) { checkedFilterMap =>
+      checkedFilterMask.map { checkedFilterMask =>
+        withResource(checkedFilterMask) { checkedFilterMask =>
           val colTypes = GpuColumnVector.extractTypes(batch)
           withResource(GpuColumnVector.from(batch)) { tbl =>
-            withResource(tbl.filter(checkedFilterMap)) { filteredData =>
+            withResource(tbl.filter(checkedFilterMask)) { filteredData =>
               GpuColumnVector.from(filteredData, colTypes)
             }
           }

--- a/tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala
+++ b/tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala
@@ -423,7 +423,7 @@ class AdaptiveQueryExecSuite
     val conf = new SparkConf()
       .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
       .set(SQLConf.LOCAL_SHUFFLE_READER_ENABLED.key, "true")
-      .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, "500")
+      .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, "400")
       .set(SQLConf.ADVISORY_PARTITION_SIZE_IN_BYTES.key, "50")
       // disable DemoteBroadcastHashJoin rule from removing BHJ due to empty partitions
       .set(SQLConf.NON_EMPTY_PARTITION_RATIO_FOR_BROADCAST_JOIN.key, "0")

